playbooks/azure/openshift-cluster/tasks/create_offer.yml

```diff
@@ -0,0 +1,59 @@
+---
+- name: ensure libraries are available
+  import_role:
+    name: lib_utils
+
+- name: set start and expire dates
+  set_fact:
+    sas_expire: "{{ lookup('pipe', 'date -d \"14 days\" +\"%Y-%m-%dT%H:%M:00Z\"') }}"
+    sas_start: "{{ lookup('pipe', 'date -d \"3 days ago\" +\"%Y-%m-%dT%H:%M:00Z\"') }}"
+    disks:
+    - "{{ image_name }}.vhd"
+
+- name: generate sas url for the container
+  command: |
+    az storage container generate-sas --name {{ openshift_azure_container }} --account-name {{ openshift_azure_storage_account }} --permissions rl --expiry {{ sas_expire }} --start {{ sas_start }} --output=json
+  register: sasurl
+
+- name: set the sas URLS
+  set_fact:
+    openshift_azure_sas_urls: "{{ openshift_azure_sas_urls|default([]) + ['https://' + openshift_azure_storage_account + '.blob.core.windows.net/' + openshift_azure_container + '/' + item + '?' + sasurl.stdout|from_json] }}"
+  with_items: "{{ disks }}"
+
+- name: fetch the current offer and update the versions
+  oo_azure_rm_publish_image_facts:
+    offer: "{{ openshift_azure_offer }}"
+  register: offerout
+
+# seeing an issue when fetching returns a 404
+# There is no Staging slot deployed for Offer ID '<offerid>', Publisher ID '<publisherid>'
```

Comment:
I'm not understanding this comment or the code - please make the comment clearer - why does a 404 happen, what do you do if it does? if it doesn't?